vault/data_source_auth_backend.go

```diff
@@ -0,0 +1,87 @@
+package vault
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func authBackendDataSource() *schema.Resource {
+	return &schema.Resource{
+		Read: authBackendDataSourceRead,
+		Schema: map[string]*schema.Schema{
+			"path": {
```

Comment:
Question, would it be possible to drop this "path" parameter and to simply record all the auths returned by the list response? I'm thinking that the advantage might be that if you needed information on _multiple_ paths, you could get them in only one call. Also, since this is a list endpoint, I wonder if getting the full response might be more what a user would _expect_, so I'm wondering whether we'd eventually get a request for that down the line. Let me know what you think.

Reply:
It is of course possible to add a `auth_backends` data source, as equivalents already exist in other providers:
- https://www.terraform.io/docs/providers/aws/d/availability_zones.html
- https://www.terraform.io/docs/providers/aws/d/availability_zone.html

The advantage of such a data source would be to avoid requesting multiple times the list of auth backends (knowing that our implem, does in fact list all of them).

We could return a map of `path` to auth backend, which should amount to the same data. However I feel like readability counts, and doing that incurs a non negligible cost:

```
data "vault_auth_backends" "mounts" {}
...
resource ".." ".." {
  mount_accessor = vault_auth_backend.mounts["oidc"].mount_accessor
} 
```
vs
```
data "vault_auth_backend" "oidc" {
  path = "oidc"
}
...
resource ".." ".." {
  mount_accessor = data.vault_auth_backend.oidc.mount_accessor
} 
```

For people iterating on the list of mounts, it will surely be useful so we can preemptively add it if you want. LMK.

Reply:
I think that seems reasonable! Thanks for talking it through with me.